.github/actions/install-gstreamer/action.yml

```
@@ -0,0 +1,127 @@
+    required: false
+  runtime-installer-path:
+    description: 'Local path of the GStreamer runtime installer'
+    required: false
+  development-installer-path:
+    description: 'Local path of the GStreamer development installer'
+    required: false
+outputs:
+  install-dir:
+    description: 'GStreamer install directory'
+    value: ${{ steps.install-gstreamer.outputs.gstreamer-install-dir }}
+  plugin-dir:
+    description: 'GStreamer plugin directory'
+    value: ${{ steps.install-gstreamer.outputs.gstreamer-plugin-dir }}
+  bin-dir:
+    description: 'GStreamer bin directory'
+    value: ${{ steps.install-gstreamer.outputs.gstreamer-bin-dir }}
+
+runs:
+  using: "composite"
+  steps:
+    # Mutual exclusive -- either set url or path for each.  Error if both or neither
+    # are set.  If path is set, verify the file exists.
+    - if: ${{ inputs.runtime-installer-url && inputs.runtime-installer-path }}
+      shell: msys2 -eo pipefail {0}
+      run: |
+        echo "Error: Cannot specify both runtime-installer-url and runtime-installer-path"
+        exit 1
+    - if: ${{ inputs.development-installer-url && inputs.development-installer-path }}
+      shell: msys2 -eo pipefail {0}
+      run: |
+        echo "Error: Cannot specify both development-installer-url and development-installer-path"
+        exit 2
+    - if: ${{ !inputs.runtime-installer-url && !inputs.runtime-installer-path }}
+      shell: msys2 -eo pipefail {0}
+      run: |
+        echo "Error: Must specify either runtime-installer-url or runtime-installer-path"
+        exit 3
+    - if: ${{ !inputs.development-installer-url && !inputs.development-installer-path }}
+      shell: msys2 -eo pipefail {0}
+      run: |
+        echo "Error: Must specify either development-installer-url or development-installer-path"
+        exit 4
+    - if: ${{ inputs.runtime-installer-path }}
+      shell: msys2 -eo pipefail {0}
+      run: |
+        if [[ ! -f "${{ inputs.runtime-installer-path }}" ]]; then
+          echo "Error: Runtime installer path does not exist: ${{ inputs.runtime-installer-path }}"
+          exit 5
+        fi
+    - if: ${{ inputs.development-installer-path }}
+      shell: msys2 -eo pipefail {0}
+      run: |
+        if [[ ! -f "${{ inputs.development-installer-path }}" ]]; then
+          echo "Error: Development installer path does not exist: ${{ inputs.development-installer-path }}"
+          exit 6
+        fi
+
+    # If the URL is provided, download the MSI, set path.
+    # This will error out if the file is missing.
+    - if: ${{ !inputs.runtime-installer-path }}
+      id: download-runtime
+      name: Download GStreamer Runtime MSI
+      shell: powershell
+      run: |
+        $msi_path = "$(( $pwd ))\gstreamer-runtime.msi"
+        Invoke-WebRequest -Uri "${{ inputs.runtime-installer-url }}" -OutFile "$msi_path"
+        echo "msi-path=$msi_path" | Out-File -FilePath $Env:GITHUB_OUTPUT -Append -Encoding utf8
```

Wouldn't hurt to annotate here:


```suggestion
        echo "msi-path=$msi_path" | Out-File -FilePath $Env:GITHUB_OUTPUT -Append -Encoding utf8
        echo "::notice title=GStreamer runtime installer downloaded::Downloaded GStreamer runtime installer from ${{ inputs.runtime-installer-url }}"
```